gobblin-modules/gobblin-kafka-09/src/main/java/gobblin/source/extractor/extract/kafka/KafkaSimpleStreamingExtractor.java

```diff
@@ -0,0 +1,209 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gobblin.source.extractor.extract.kafka;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.hadoop.util.hash.Hash;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+
+import com.google.common.base.Preconditions;
+import com.google.gson.JsonElement;
+
+import kafka.Kafka;
+import kafka.common.TopicAndPartition;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.WorkUnitState;
+import gobblin.configuration.State;
+import gobblin.metrics.kafka.KafkaSchemaRegistry;
+import gobblin.metrics.kafka.SchemaRegistryException;
+import gobblin.source.extractor.DataRecordException;
+import gobblin.source.extractor.Extractor;
+import gobblin.source.extractor.Watermark;
+import gobblin.source.extractor.extract.EventBasedExtractor;
+import gobblin.source.extractor.StreamingExtractor;
+import gobblin.source.extractor.RecordEnvelope;
+import gobblin.source.extractor.CheckpointableWatermark;
+import gobblin.source.extractor.ComparableWatermark;
+import gobblin.source.extractor.extract.LongWatermark;
+import gobblin.metrics.Tag;
+import gobblin.source.extractor.extract.kafka.SimpleKafkaSchemaRegistry;
+import gobblin.writer.WatermarkStorage;
+
+
+/**
+ * An implementation of {@link StreamingExtractor} from which reads from Kafka and returns records as an array of bytes.
+ *
+ * @author Shrikanth Shankar
+ *
+ *
+ */
+public class KafkaSimpleStreamingExtractor extends EventBasedExtractor<String, RecordEnvelope<byte[]>> implements StreamingExtractor<String, byte[]>, WatermarkStorage{
+
+    public static class KafkaWatermark implements CheckpointableWatermark {
+      TopicPartition _topicPartition;
+      LongWatermark _lwm;
+
+      KafkaWatermark(TopicPartition topicPartition, LongWatermark lwm) {
+        _topicPartition = topicPartition;
+        _lwm = lwm;
+      }
+
+      @Override
+	    public String getSource() {
+	      return _topicPartition.toString();
+	    }
+
+      @Override
+	    public ComparableWatermark getWatermark() {
+	      return _lwm;
+	    }
+
+      @Override
+      public short calculatePercentCompletion(Watermark lowWatermark, Watermark highWatermark) {
+        return 0;
+      }
+
+      @Override
+      public JsonElement toJson() {
```

Comment:
This is important to implement if this Watermark gets persisted in the state store. If you don't implement it now, leave a TODO comment to fix it.